modules/data.atmosphere/R/tdm_temporal_downscale.R

```diff
@@ -0,0 +1,656 @@
+                "next.air_temperature_min", "next.precipitation_flux", 
+                "next.surface_downwelling_shortwave_flux_in_air", "next.surface_downwelling_longwave_flux_in_air", 
+                "next.air_pressure", "next.specific_humidity", "next.wind_speed")]
+            dat.temp$surface_downwelling_longwave_flux_in_air <- 99999  # Dummy value so there's a column
+            day.now <- unique(dat.temp$doy)
+            
+            # Set up the lags
+            if (i == min(dat.mod$time.day)) {
+                sim.lag <- stack(lags.init$surface_downwelling_longwave_flux_in_air)
+                names(sim.lag) <- c("lag.surface_downwelling_longwave_flux_in_air", 
+                  "ens")
+                
+            } else {
+                sim.lag <- stack(data.frame(array(dat.sim[["surface_downwelling_longwave_flux_in_air"]][dat.mod$time.day == 
+                  (i - 1) & dat.mod$hour == 23, ], dim = c(1, ncol(dat.sim$surface_downwelling_longwave_flux_in_air)))))
+                names(sim.lag) <- c("lag.surface_downwelling_longwave_flux_in_air", 
+                  "ens")
+            }
+            dat.temp <- merge(dat.temp, sim.lag, all.x = T)
+            
+            # dat.temp$surface_downwelling_shortwave_flux_in_air <-
+            # stack(dat.sim$surface_downwelling_shortwave_flux_in_air[rows.now,])[,1]
+            
+            
+            # Load the saved model
+            load(file.path(path.model, "surface_downwelling_longwave_flux_in_air", 
+                paste0("model_surface_downwelling_longwave_flux_in_air_", 
+                  day.now, ".Rdata")))
+            mod.surface_downwelling_longwave_flux_in_air.doy <- mod.save
+            rm(mod.save)
+            
+            
+            # Pull the coefficients (betas) from our saved matrix
+            rows.beta <- sample(1:n.beta, n.ens, replace = T)
+            betas.surface_downwelling_longwave_flux_in_air <- ncdf4::nc_open(file.path(path.model, 
+                "surface_downwelling_longwave_flux_in_air", "betas_surface_downwelling_longwave_flux_in_air_1.nc"))
+            Rbeta <- as.matrix(ncdf4::ncvar_get(betas.surface_downwelling_longwave_flux_in_air, 
+                paste(day.now))[rows.beta, ], nrow = length(rows.beta), 
+                ncol = ncol(betas))
+            ncdf4::nc_close(betas.surface_downwelling_longwave_flux_in_air)
+            
+            dat.pred <- predict.met(newdata = dat.temp, model.predict = mod.surface_downwelling_longwave_flux_in_air.doy, 
+                Rbeta = Rbeta, resid.err = F, model.resid = NULL, Rbeta.resid = NULL, 
+                n.ens = n.ens)
+            dat.pred <- dat.pred^2  # because squared to prevent negative numbers
+            
+            # Hard-coding some sanity bounds by ball-parking things from NLDAS &
+            # CRUNCEP This is necessary if you have poorly constrained training
+            # models
+            dat.pred[dat.pred < 100] <- 100
```

Comment:
at night wouldnt you expect radiation == 0? Or in artic in winter?

Is this the best place to have sanity checks like this? Or would it be possible to compare the input data with the downscaled data as sanity check (e.g. total radiation, precip, etc)